question-service/src/service/question-service.ts

```diff
@@ -8,18 +8,31 @@ export async function createQuestion(questionData: any) {
 }
 
 export async function fetchAllQuestions(reqQuery: any) {
-    const {sort = 'title', order = 'asc', page = 1, limit = 10 } = reqQuery;
+    const {sort = 'title', order = 'asc', page = 1, limit = 10 , search,} = reqQuery;
 
+    // Build filter object
+    let filter: any = {};
+
+    if (search) {
+        const searchRegex = new RegExp(search, 'i'); // Case-insensitive search
+    
+        filter.$or = [
+          { title: { $regex: searchRegex } },
+          { category: { $regex: searchRegex } },
```

Comment:
@CedricChia123 am still filtering the topic in search query, will remove if not required